### PR TITLE
chore: simplify input of blackduck scan workflow

### PR DESCRIPTION
**Pull request by wattachai-lseg**
## Description

simplify input of blackduck scan workflow

Fixes # https://jira.refinitiv.com/browse/ELF-2321

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .github/workflows/blackduck_scan.yml [hunk 1]
```diff
@@ -55,9 +50,13 @@ jobs:
           curl https://sig-repo.synopsys.com/artifactory/bds-integrations-release/com/synopsys/integration/synopsys-detect/${{ vars.SYNOPSYS_DETECT_VERSION }}/synopsys-detect-${{ vars.SYNOPSYS_DETECT_VERSION }}.jar -o $DETECT_FILE
           mkdir appsec || true
           mv $DETECT_FILE appsec/
+          BRANCH_NAME="${GITHUB_REF#refs/heads/}"
 
-          if [[ "${{ inputs.production }}" == "true" ]]; then  export RELEASE_TYPE="PROD"; export VERSION_ID="Release_${GITHUB_REF#refs/heads/}"; fi
-          if [[ "${{ inputs.production }}" == "false" ]]; then export RELEASE_TYPE="DEV"; export VERSION_ID="Development-${{ inputs.version }}"; fi
+          if [[ "${{ inputs.target }}" == "Production" && $BRANCH_NAME != "v6" && $BRANCH_NAME != "v7" ]]; \
```
Comment (Sakchai-Refinitiv) [inline .github/workflows/blackduck_scan.yml]: The validation should separate to the step after checkout

File: .github/workflows/blackduck_scan.yml [hunk 2]
```diff
@@ -55,9 +50,13 @@ jobs:
...
+          then echo "Only v6 & v7 branch can be published as production"; exit 1; \
+          fi
+          if [[ "${{ inputs.target }}" == "Production" ]]; then export RELEASE_TYPE="PROD"; export VERSION_ID="Release_$BRANCH_NAME"; fi
+          if [[ "${{ inputs.target }}" != "Production" ]]; then export RELEASE_TYPE="DEV"; export VERSION_ID="${{ inputs.target }}"; fi
```
Comment (Sakchai-Refinitiv) [inline .github/workflows/blackduck_scan.yml]: We can reduce the two if conditions to one, by declare the default variables for development and then override them to production in the if condition.
```bash
RELEASE_TYPE="DEV"
VERSION_ID="${{ inputs.target }}"

[[ "${{ inputs.target }}" == "Production" ]] && RELEASE_TYPE="PROD" && VERSION_ID="Release_$BRANCH_NAME";
```
Note. export might be removed if it use for CLI only

File: .github/workflows/blackduck_scan.yml (hunk 1 above)
Comment (wattachai-lseg) [reply .github/workflows/blackduck_scan.yml]: updated

File: .github/workflows/blackduck_scan.yml (hunk 2 above)
Comment (wattachai-lseg) [reply .github/workflows/blackduck_scan.yml]: updated

**Review by Sakchai-Refinitiv — APPROVED** [review]
Nice 👍

**Pull request merged**